fmf/utils.py

```diff
@@ -321,7 +331,7 @@ def check_clause(clause):
 
     # At least one clause must be true
     return any([check_clause(clause)
-                for clause in re.split(r"\s*\|\s*", filter)])
+                for clause in re.split(r"\s*\|\s*(?=[^|]*:)", filter)])
```

Comment:
@The-Mule Could you explain why is this change necessary, this isn't just PR for docs then.

Reply:
Uh oh, good catch, this slipped in by mistake, sorry - this was bit related to https://github.com/teemtee/fmf/pull/219#issuecomment-2091219335, I'll remove it.

Reply:
Fixed.